File: JavaScript/프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    var answer = [0, 0];
    let turn = 0;
    let person;
    const newObj = {};
    let lastWord = " ";
    for (let [i, word] of words.entries()){
        if (i % n === 0) turn += 1;
        if (lastWord !== " "){
            if (newObj[word] === 1 || lastWord !== word[0]){
                person = (i % n) + 1;
                break;
            }
        }
        
        lastWord = word[word.length - 1];
        newObj[word] = 1;
    }
    if (person === undefined) return answer;
    answer = [person, turn]
    return answer;
}

File: C++/프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer;
    map<string, int> mp;
    answer = {0, 0};
    char last_word;
    for (int i = 0; i < words.size(); i++){
        if (mp.count(words[i])) {
            answer = {(i % n) + 1, i / n + 1};
            break;
        }
        if (i == 0){
            last_word = words[0][words[0].size() - 1];   
        } else {
            if (last_word != words[i][0]) {
                answer = {(i % n) + 1, i / n + 1};
                break;
            }
            last_word = words[i][words[i].size() - 1]; 
        }
        mp[words[i]] = 1;
    }

    return answer;
}